tmt/steps/discover/__init__.py

```diff
@@ -94,7 +104,7 @@ def wake(self):
 
         # Choose the right plugin and wake it up
         for data in self.data:
-            plugin = DiscoverPlugin.delegate(self, data)
+            plugin = DiscoverPlugin.delegate(self, dict(data))
```

Comment:
hmm, isn't really type annotation of `delegate` function incorrect?

It is now

```
  445     @classmethod
  446     def delegate(
  447             cls,
  448             step: Step,
  449             data: Dict[Any, Any]) -> 'Plugin':
```

shouldn't `data` be rather have `StepData` type?

Reply:
Fyi, there's also a discussion about this same line in an another PR: https://github.com/teemtee/tmt/pull/1319#discussion_r916599351

After looking briefly at other `delegate()` method calls, it seems that changing the type to `StepData` would indeed be more suitable.

Reply:
Yes, this should be covered by ffaa8ec303e33156f4517a5dc9bad7f3cf7d11b5.